server/aws-lsp-codewhisperer/src/language-server/agenticChat/tools/mcp/mcpManager.ts

```
@@ -165,7 +165,17 @@ export class McpManager {
                     )
                     timer.unref()
                 })
-                await Promise.race([connectPromise, timeoutPromise])
+                try {
+                    await Promise.race([connectPromise, timeoutPromise])
```

I think what we should do is not catch on the race, instead just try catch client.connect(transport), then add our own message, also, we should add the actual error message in the error. 

something like this 

```
const connectPromise = client.connect(transport).catch(err => {
    throw new AgenticChatError(
        `MCP: server '${serverName}' failed to connect: ${err.message}`,
        'MCPServerConnectionFailed'
    )
})
```

this way we also don't swallow the init timeout error.